### PR TITLE
21: validate template matches expected columns

### LINKED ISSUE #21: Validate upload
As I provider, if I try to upload a CSV or Excel file that is not formatted according to the template, or if I try to upload a file that is not a CSV or Excel file, I get an error that I must upload a CSV or Excel file formatted according to the template.

First pass - check ~file type and~ headers

### PR DESCRIPTION
**Pull request by jlhogan**
closes #21

### CODE REVIEW

File: src/entity/FlattenedEnrollment.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ export class FlattenedEnrollment implements FlattenedEnrollmentInterface {
```
Comment (jlhogan) [inline src/entity/FlattenedEnrollment.ts:173]: changes here are to match the checked in template until (or if ever) template is generated from the model

File: src/entity/FlattenedEnrollment.ts [hunk 2]
```diff
@@ -356,7 +356,7 @@ export class FlattenedEnrollment implements FlattenedEnrollmentInterface {
     formattedName: 'Provider',
     required: REQUIRED,
     definition: 'The provider from which the child is receiving services.',
-    reason: 'Used to link child information to provider data.',
+    reason: 'Used to link child information to provi data.',
```
Comment (jlhogan) [inline src/entity/FlattenedEnrollment.ts:359]: typo

File: src/entity/decorators/dataDefinition.ts [hunk 3]
```diff
@@ -22,7 +23,18 @@ export const UTILIZATION_REPORTING_REASON =
...
+  required: string;
+  definition: string;
+  reason: string;
+  format: string;
+  example: string;
+  section: string;
+  propertyName: string;
+}
+
+type DataDefinitionInfoInput = {
```
Comment (jlhogan) [inline src/entity/decorators/dataDefinition.ts:37]: allow the user to set data defn info without providing propertyName, which should come from the model 
see `getDataDefinition` below

todo: add comment

File: src/entity/FlattenedEnrollment.ts [hunk 4]
```diff
@@ -464,7 +464,7 @@ export class FlattenedEnrollment implements FlattenedEnrollmentInterface {
 
   @Column({ nullable: true })
   @DataDefinition({
-    formattedName: 'First funding reporting period',
+    formattedName: 'First funding period',
```
Comment (cailyncodes) [inline src/entity/FlattenedEnrollment.ts:467]: Looks like every other column is every word caps, but this is only sentence case. Just confirming this is intended

File: src/entity/FlattenedEnrollment.ts [hunk 5]
```diff
@@ -477,7 +477,7 @@ export class FlattenedEnrollment implements FlattenedEnrollmentInterface {
 
   @Column({ nullable: true })
   @DataDefinition({
-    formattedName: 'Last funding reporting period',
+    formattedName: 'Last funding period',
```
Comment (cailyncodes) [inline src/entity/FlattenedEnrollment.ts:480]: Same here

File: src/routes/enrollmentReport.ts [hunk 6]
```diff
@@ -37,12 +37,29 @@ router.post(
...
-      })
-    );
-    res.send(report);
+    try {
+      const { EXPECTED_HEADERS, headers, enrollments } = parseUploadedTemplate(
+        req.file
+      );
+
+      EXPECTED_HEADERS.forEach((header, idx) => {
+        if (header !== headers[idx]) console.log(header, headers[idx]);
```
Comment (cailyncodes) [inline src/routes/enrollmentReport.ts:46]: Do we want this console.log ?

File: src/utils/parseUploadedTemplate.ts [hunk 7]
```diff
@@ -1,9 +1,15 @@
 import { readFile, utils, WorkSheet } from 'xlsx';
 import { getConnection, getManager } from 'typeorm';
 import { FlattenedEnrollment } from '../entity';
+import { getDataDefinition } from '../entity/decorators/dataDefinition';
 
 /**
- * Parses the uploaded file into an array of FlattenedEnrollments
+ * Parses the uploaded file into: 
+ * 	- an array of column header strings from the uploaded template
```
Comment (cailyncodes) [inline src/utils/parseUploadedTemplate.ts:8]: Nit: spacing

File: src/entity/FlattenedEnrollment.ts (hunk 4 above)
Comment (jlhogan) [reply src/entity/FlattenedEnrollment.ts:467]: yeah no nothing about the way these titles do (rather do not) match is intentional but fixing the templates is annoying so i am just leaving them as is until we update the BE to serve generated templates from this metadata. At that point we should fix these to be consistently sentence or title case :)

File: src/routes/enrollmentReport.ts (hunk 6 above)
Comment (jlhogan) [reply src/routes/enrollmentReport.ts:46]: nope!

**Pull request merged**